src/actions/measurements.js

```diff
@@ -0,0 +1,121 @@
+  if (!collectionKey) return collections[0];
+  const potentialCollections = collections.filter((collection) => collection.key === collectionKey);
+  if (potentialCollections.length === 0) return collections[0];
+  if (potentialCollections.length > 1) {
+    console.error(`Found multiple collections with key ${collectionKey}. Returning the first matching collection only.`);
+  }
+  return potentialCollections[0];
+};
+
+/**
+ * Constructs the controls redux state for the measurements panel based on
+ * config values within the provided collection.
+ *
+ * If no display defaults are provided, uses the current controls redux state.
+ * If the current `measurementsGrouping` does not exist in the collection, then
+ * defaults to the first grouping option.
+ * @param {Object} collection
+ * @returns {Object}
+ */
+const getCollectionDisplayControls = (controls, collection) => {
+  const measurementsControls = [
+    "measurementsGroupBy",
+    "measurementsDisplay",
+    "measurementsShowOverallMean",
+    "measurementsShowThreshold"
+  ];
+  // Copy current control options for measurements
+  const newControls = pick(controls, measurementsControls);
+  // Checks the current group by is available as a field in collection
+  if (!collection.fields.has(newControls.measurementsGroupBy)) {
+    // If current group by is not available as a field, then default to the first grouping option.
+    newControls.measurementsGroupBy = collection.groupings[0];
+  }
+
+  if (collection["display_defaults"]) {
+    const {
+      group_by,
+      measurement_display,
+      show_overall_mean,
+      show_threshold
+    } = collection["display_defaults"];
+
+    if (group_by) {
+      newControls.measurementsGroupBy = group_by;
+    }
+    if (measurement_display) {
+      newControls.measurementsDisplay = measurement_display;
+    }
+    if (typeof show_overall_mean === "boolean") {
+      newControls.measurementsShowOverallMean = show_overall_mean;
+    }
+    if (typeof show_threshold === "boolean") {
+      newControls.measurementsShowThreshold = show_threshold;
+    }
+  }
+
+  return newControls;
+};
+
+export const loadMeasurements = (json) => (dispatch, getState) => {
+  // TODO: Load controls from state to get potential url query parameters
+  const { tree, controls } = getState();
+  if (!tree.loaded) {
+    throw new Error("tree not loaded");
+  }
+
+  const collections = json["collections"];
+  if (!collections || collections.length === 0) {
+    throw new Error("Measurements JSON does not have collections");
+  }
+
+  collections.forEach((collection) => {
+    // Convert provided fields array to Map for easier access and to keep ordering
+    const collectionFieldsArray = collection.fields;
+    collection.fields = new Map();
+    if (collectionFieldsArray && collectionFieldsArray.length > 0) {
+      collectionFieldsArray.forEach(({key, title}) => collection.fields.set(key, {title: title || key}));
```

Comment:
Ah yeah, I debated on this data structure because it can seem overly complicated. 

My reasoning is that `collection.fields` and `collection.filters` are Maps that utilize both the keys and the values as data stores. For example, I use `collection.filters` to check whether a field should be a filter and I think the current structure makes more sense:
```js
// current
collection.filters.has(field)

// suggestion
collection.filterValues.has(field)
```